### PR TITLE
[PA][Python]add the chatting sample

### PR DESCRIPTION
**Pull request by l17813182030**
add the chatting sample

### CODE REVIEW

File: python/main.py [hunk 1]
```diff
@@ -332,9 +335,361 @@ def recognized(evt):
...
+    finally:
+        stream.close()
+
+
+def chatting_from_file():
+    """Performs pronunciation assessment asynchronously from an audio file and enable prosody/content assessment
+        with continuous mode.
+        See more information at https://aka.ms/csspeech/pa"""
+
+    topic = "YourOwnTopic"
```
Comment (wangkenpu) [inline python/main.py]: You have specified the input file. So please change `YourOwnTopic` to the real topic.

File: python/main.py [hunk 2]
```diff
@@ -332,9 +335,361 @@ def recognized(evt):
...
+
+        # Start continuous speech recognition
+        speech_recognizer.start_continuous_recognition()
+        while not done:
+            time.sleep(.5)
+
+        speech_recognizer.stop_continuous_recognition()
+        text = "".join(result_text)
+        print("YOU: ", text)
+        return text
```
Comment (wangkenpu) [inline python/main.py]: Please keep only 1 blank line.

File: python/main.py [hunk 3]
```diff
@@ -332,9 +335,361 @@ def recognized(evt):
...
+
+
+    def call_gpt(send_text):
+        your_resource_name = "YourResourceName"
```
Comment (wangkenpu) [inline python/main.py]: Please add the corresponding doc in the readme file.

**Review by wangkenpu — CHANGES_REQUESTED** [review]

File: python/main.py (hunk 3 above)
Comment (l17813182030) [reply python/main.py]: added

File: python/main.py (hunk 2 above)
Comment (l17813182030) [reply python/main.py]: updated

File: python/main.py (hunk 1 above)
Comment (l17813182030) [reply python/main.py]: have changed to sample topic

File: python/README.md [hunk 4]
```diff
@@ -40,6 +40,10 @@ This sample demonstrates various forms of pronunciation assessment using the Spe
 * [Download the sample code to your development PC.](/README.md#get-the-samples)
 * To tailor the sample to your configuration, use search and replace across the whole sample directory to update the following strings:
 
+  * `YourResourceName`: replace with your Azure resource name.
```
Comment (wangkenpu) [inline python/README.md]: Please put these after `ScenarioId`.

File: python/chatting_sample.py [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+            nonlocal display_text, mis_pronunciation_words
+            results.append(pronunciation_result)
+            recognized_words += pronunciation_result.words
+            fluency_scores.append(pronunciation_result.fluency_score)
+            json_result = evt.result.properties.get(speechsdk.PropertyId.SpeechServiceResponse_JsonResult)
+            jo = json.loads(json_result)
+            nb = jo["NBest"][0]
+            display_text += nb["Display"]
+            json_words += nb["Words"]
+            if "ProsodyScore" in nb["PronunciationAssessment"]:
```
Comment (wangkenpu) [inline python/chatting_sample.py]: Please do not get prosody score from json.

File: python/chatting_sample.py [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+    def tts(text, output_filename):
+        file_config = speechsdk.audio.AudioOutputConfig(filename=output_filename)
+        speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+        speech_synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=file_config)
+
+        result = speech_synthesizer.speak_ssml_async(text).get()
+        # Check result
+        if result.reason == speechsdk.ResultReason.SynthesizingAudioCompleted:
+            pass
```
Comment (wangkenpu) [inline python/chatting_sample.py]: Here can add `print(f"Save synthesized waveform to {output_filename}")` to let the users know where the wave is saved.

File: python/chatting_sample.py [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+        speech_recognizer.stop_continuous_recognition()
+        text = "".join(result_text)
+        print("YOU: ", text)
+        return text
+
+    def call_gpt(send_text):
+        your_resource_name = "YourResourceName"
+        your_deployment_id = "YourDeploymentId"
+        api_version = "YourApiVersion"
+        your_api_key = "YourApiKey"
```
Comment (wangkenpu) [inline python/chatting_sample.py]: Please put these after line 21.

File: python/README.md (hunk 4 above)
Comment (l17813182030) [reply python/README.md]: updated

File: python/chatting_sample.py (hunk 5 above)
Comment (l17813182030) [reply python/chatting_sample.py]: updated

File: python/chatting_sample.py (hunk 6 above)
Comment (l17813182030) [reply python/chatting_sample.py]: > Here can add `print(f"Save synthesized waveform to {output_filename}")` to let the users know where the wave is saved.

added

File: python/chatting_sample.py (hunk 7 above)
Comment (l17813182030) [reply python/chatting_sample.py]: > Please put these after line 21.

updated

File: python/README.md [hunk 8]
```diff
@@ -44,6 +44,10 @@ This sample demonstrates various forms of pronunciation assessment using the Spe
   * `YourServiceRegion`: replace with the [region](https://aka.ms/csspeech/region) your subscription is associated with.
     For example, `westus` or `westeurope`.
   * `ScenarioId`: scenario ID will be assigned by product team ("" is the default value for the general model)
+  * `YourResourceName`: replace with your Azure resource name.
```
Comment (wangkenpu) [inline python/README.md:47]: At the section "Build the sample", please change `pip install azure-cognitiveservices-speech` to `pip install azure-cognitiveservices-speech requests`

File: python/README.md [hunk 9]
```diff
@@ -44,6 +44,10 @@ This sample demonstrates various forms of pronunciation assessment using the Spe
...
+  * `YourDeploymentId`: replace with your Azure OpenAi deployment name.
```
Comment (wangkenpu) [inline python/README.md]: `OpenAi` -> `OpenAI`

File: python/README.md [hunk 10]
```diff
@@ -44,6 +44,10 @@ This sample demonstrates various forms of pronunciation assessment using the Spe
...
+  * `YourApiVersion`: replace with the [api version](https://learn.microsoft.com/en-US/azure/ai-services/openai/reference).
```
Comment (wangkenpu) [inline python/README.md]: `api` -> `API`

File: python/chatting_sample.py [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+    Importing the Speech SDK for Python failed.
+    Please run "pip install azure-cognitiveservices-speech" to install the SDK.
+    """)
+    sys.exit(1)
+
+
+# Set up the subscription info for the Speech Service:
+# Replace with your own subscription key and service region (e.g., "westus").
+speech_key, service_region = "YourSubscriptionKey", "YourServiceRegion"
+your_resource_name = "YourResourceName"
```
Comment (wangkenpu) [inline python/chatting_sample.py:26]: Add `Set up the parameters for Azure OpenAI Services` before this line.

File: python/chatting_sample.py [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+        stream.close()
+
+
+def chatting_from_file():
+    """Performs pronunciation assessment asynchronously from an audio file and enable prosody/content assessment
+        with continuous mode.
+        See more information at https://aka.ms/csspeech/pa"""
+
+    topic = "Talk about your day today"
+    input_files = ["../resources/chat_input_1.wav", "../resources/chat_input_2.wav"]
```
Comment (wangkenpu) [inline python/chatting_sample.py:104]: Can you change the input wavefiles to our demo waveform?

File: python/chatting_sample.py [hunk 13]
```diff
@@ -0,0 +1,390 @@
...
+
+            if scores_dict["fluency score"] < 60 or scores_dict["prosody score"] < 60:
+                if scores_dict["fluency score"] < 60:
+                    origin_content += "Fluency "
+                if scores_dict["prosody score"] < 60:
+                    origin_content += "Prosody "
+                origin_content += "report: "
+                origin_content += get_error_message(["Missing break", "Unexpected break", "Monotone"])
+
+            tts(f'<speak version="1.0" xmlns="http://www.w3.org/2001/10/synthesis" xml:lang="en-US">\
```
Comment (wangkenpu) [inline python/chatting_sample.py]: Please check this output. I run this sample but the output wave is illegal and I can not play it.

File: python/chatting_sample.py [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+            for score_key in scores_dict:
+                set_message_dict(scores_dict[score_key], score_key)
+
+            messages = ""
+            for message_key in message_dict:
+                if message_dict[message_key] != []:
+                    is_or_are = "is" if len(message_dict[message_key]) == 1 else "are"
+                    messages += f"{' and '.join(message_dict[message_key])} {is_or_are} {message_key}. "
+
+            tts((
```
Comment (wangkenpu) [inline python/chatting_sample.py]: too many "and" in the score comment output.

**Review by wangkenpu — CHANGES_REQUESTED** [review]

File: python/README.md (hunk 8 above)
Comment (l17813182030) [reply python/README.md:47]: added

> At the section "Build the sample", please change `pip install azure-cognitiveservices-speech` to `pip install azure-cognitiveservices-speech requests`

File: python/README.md (hunk 9 above)
Comment (l17813182030) [reply python/README.md]: > `OpenAi` -> `OpenAI`

ok changed

File: python/README.md (hunk 10 above)
Comment (l17813182030) [reply python/README.md]: > `api` -> `API`

changed

File: python/chatting_sample.py (hunk 11 above)
Comment (l17813182030) [reply python/chatting_sample.py:26]: > Add `Set up the parameters for Azure OpenAI Services` before this line.

ok, added

File: python/chatting_sample.py (hunk 12 above)
Comment (l17813182030) [reply python/chatting_sample.py:104]: > Can you change the input wavefiles to our demo waveform?

changed

File: python/chatting_sample.py (hunk 14 above)
Comment (l17813182030) [reply python/chatting_sample.py]: > too many "and" in the score comment output.

I changed "and" to "," just like in the studio

Comment (wangkenpu) [reply python/chatting_sample.py:380]: Please add "and" for the last dimension.

**Review by wangkenpu — APPROVED** [review]

**Pull request merged**